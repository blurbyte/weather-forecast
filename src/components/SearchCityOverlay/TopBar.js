import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import { Logo } from '../Icons';
import CloseButton from './CloseButton';

const Wrapper = styled.header`
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: #222222;
  color: #ffffff;
  height: 6rem;
  padding: 0 2rem;
`;

const TopBar = ({ onClose }) => (
  <Wrapper onClick={onClose}>
    <Logo />
    <CloseButton />
  </Wrapper>
);

TopBar.propTypes = {
  onClose: PropTypes.func.isRequired
};

export default TopBar;
