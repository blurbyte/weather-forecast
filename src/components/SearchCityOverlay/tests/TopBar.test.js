import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import TopBar from '../TopBar';

jest.mock('../../Icons', () => ({
  Logo: 'Logo'
}));

jest.mock('../CloseButton', () => 'CloseButton');

test('renders correctly', () => {
  const component = renderer.create(<TopBar onClose={jest.fn()} />);
  expect(component.toJSON()).toMatchSnapshot();
});
