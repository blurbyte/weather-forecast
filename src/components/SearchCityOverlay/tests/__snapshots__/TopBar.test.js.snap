// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background-color: #222222;
  color: #ffffff;
  height: 6rem;
  padding: 0 2rem;
}

<header
  className="c0"
  onClick={[Function]}
>
  <Logo />
  <CloseButton />
</header>
`;
